tom/packages.py

```diff
@@ -0,0 +1,175 @@
+import re
+import json
+import requests
+import collections
+import datetime
+from tom.git import GitRepo
+
+class PackageMapperException(Exception):
+    """Base class for all exceptions in this file"""
+    pass
+
+class RepoFileNotFoundException(PackageMapperException):
+    """Exception that is risen if a file was not found in repo"""
+    pass
+
+class URLDownloadFailureException(PackageMapperException):
+    """Exception that is risen if a file was not downloaded"""
+    pass
+
+class JSONParsingError(PackageMapperException):
+    """Exception that is risen when a JSON file is not a valid JSON"""
+    pass
+
+class JSONStructureError(PackageMapperException):
+    """Exception that is risen when JSON doesn't have expected fields"""
+    pass
+
+def is_branch_or_version(string):
+    '''Tries to figure out if passed argument is branch or version.
+    Returns 'branch', 'version', or False if deduction failed.
+    Branch is either 'master' or something like 3.12.x;
+    version is something like 3.12.5,
+    optionally followed by letter (3.12.5b) for aplha/beta/gamma...zeta,
+    optionally followed by release (3.12.5-2).
+    '''
+    if string == 'master' or re.match('3\.\\d+\.x$', string):
+        return 'branch'
+    if re.match('3\\.\\d+\\.\\d+[a-z]?(-\\d+)?$', string):
+        return 'version'
+    return None
```

Comment:
Should this maybe raise an exception instead? We wouldn't need the `assert` then.